### PR TITLE
Add modules and API for easy training pipeline

### LINKED ISSUE #177: Training of FLAMO
Build convenient Ecosystem
- Initalizations
- Loss functions
- Training Pipeline

Compare with Orchi's and Jeremy's training scripts.

### PR DESCRIPTION
**Pull request by jeremybbq**
## Summary
Add modules and API for easy training pipeline, resolves #177.

**Revisiting the training setup in FLAMO, especially the part using Dataset. A lighter optimizer that is not stochastic might be possible.**

## Changes
- scripts in `src/train`
- new example for colorless FDN training

## Testing
- [ ] `pytest` passes locally
- [ ] New tests added for new functionality
- [ ] Coverage not decreased (`pytest --cov=pyFDN`)

## Checklist
- [ ] Pre-commit hooks pass (`pre-commit run --all-files`)
- [ ] Docstrings updated for any changed public API
- [ ] `HISTORY.rst` updated if this is a user-facing change

### CODE REVIEW

File: src/pyFDN/generate/sample_delay_lengths.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    if N < 1:
+        raise ValueError("N must be positive")
+    low, high = delay_range
+    if low < 1 or high <= low:
+        raise ValueError("delay_range must satisfy 1 <= low < high")
+    if sigma is not None and sigma <= 0:
+        raise ValueError("sigma must be positive")
+
+    generator = _as_generator(rng)
+    targets = _sample(N, low, high, distribution, generator)
```
[the remark this answers is not in the document]
Comment (jeremybbq) [reply src/pyFDN/generate/sample_delay_lengths.py]: removed sigma

File: src/pyFDN/train/extract.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+"""Step 4 of training: read a trained flamo model back into an FDNBuild."""
```
[the remark this answers is not in the document]
Comment (jeremybbq) [reply src/pyFDN/train/extract.py]: fixed

File: src/pyFDN/train/extract.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return FDNBuild(
+        A=p.A,
+        B=p.B,
+        C=p.C,
+        D=p.D,
+        delays=p.delays,
+        fs=float(p.fs if p.fs is not None else (fs or 0.0)),
+        filters=p.attenuation_sos,
+        post_eq=p.post_eq_sos,
+    )
```
[the remark this answers is not in the document]
Comment (jeremybbq) [reply src/pyFDN/train/extract.py]: made sample rate hard requirement

File: examples/example_train_colorless_FDN.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+@app.cell
+def _(np, pyFDN):
+    fs = 48000
+    nfft = 2**11
+
+    # 1. build a small "tiny colorless" lossless skeleton: 8 coprime delays.
+    delays = pyFDN.sample_delay_lengths(
+        8, (800, 3200), distribution="uniform", coprime=False, rng=0
+    )
```
[the remark this answers is not in the document]
Comment (jeremybbq) [reply examples/example_train_colorless_FDN.py]: fixed comments

File: src/pyFDN/train/objectives.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    # Shape target to (1, rows, n_out): rows = its own length for a magnitude
+    # target, or nfft (zero-padded/truncated) for a time target; a single-channel
+    # target is broadcast across outputs.
+    arr = np.asarray(target, dtype=np.float64)
+    if arr.ndim == 1:
+        arr = arr[:, None]
+    if arr.shape[1] != n_out:
+        arr = np.repeat(arr[:, :1], n_out, axis=1)
+    rows = arr.shape[0] if target_kind == "magnitude" else nfft
+    buf = np.zeros((rows, n_out))
```
[the remark this answers is not in the document]
Comment (jeremybbq) [reply src/pyFDN/train/objectives.py]: fixed
